### PR TITLE
Support for Babel transformed es6 classes

### PR DESCRIPTION
**Pull request by geoffdutton**
I noticed the spyAll() method wasn't worked for es6 classes that are transformed by babel.

### CODE REVIEW

**Comment by geoffdutton** [comment]
I need to look for obj.__proto__ for es6 babel classes since the babel transformation doesn't add to the obj.prototype

**Comment by geoffdutton** [comment]
After doing some more research, I ended up using Object.getPrototypeOf if it exists per the [Mozilla docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/proto):

> Warning: While Object.prototype.__proto__ is supported today in most browsers, its existence and exact behavior has only been standardized in the ECMAScript 6 specification as a legacy feature to ensure compatibility for web browsers. For better support, it is recommended that only [Object.getPrototypeOf()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getPrototypeOf) be used instead.

File: bower.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "jasmine-utils",
-  "version": "0.2.3",
+  "version": "0.2.9",
```
Comment (mjeanroy) [inline bower.json]: Can you remove this change please (the version number will be updated when the release will be made)?

File: package.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "jasmine-utils",
   "author": "mickael.jeanroy@gmail.com",
   "main": "src/jasmine-utils.js",
-  "version": "0.2.3",
+  "version": "0.2.8",
```
Comment (mjeanroy) [inline package.json]: Same comment here, can you remove this change please?

File: package.json [hunk 3]
```diff
@@ -12,19 +12,22 @@
     "jasmine-core": ">=1.3.0"
   },
   "devDependencies": {
+    "babel-preset-es2015": "^6.18.0",
```
Comment (mjeanroy) [inline package.json]: I prefer to use exact version, can we change that?

File: package.json [hunk 4]
```diff
@@ -12,19 +12,22 @@
...
     "bower": "1.7.7",
+    "gulp": "3.9.1",
+    "gulp-babel": "^6.1.2",
```
Comment (mjeanroy) [inline package.json]: Same comment here: I prefer to use exact version, can we change that?

File: package.json [hunk 5]
```diff
@@ -12,19 +12,22 @@
...
+    "gulp-bump": "1.0.0",
+    "gulp-filter": "3.0.1",
+    "gulp-git": "1.7.0",
+    "gulp-jshint": "2.0.0",
+    "gulp-tag-version": "1.3.0",
     "jasmine-core": ">=1.3.0",
+    "jshint": "2.9.1",
     "karma": "0.13.21",
+    "karma-chrome-launcher": "^2.0.0",
```
Comment (mjeanroy) [inline package.json]: Same comment here: I prefer to use exact version, can we change that?

File: src/jasmine-utils.js [hunk 6]
```diff
@@ -492,6 +492,13 @@
       };
     },
 
+    toContainIgnoringCase: function(string) {
```
Comment (mjeanroy) [inline src/jasmine-utils.js]: Can you open a new PR for that?

**Comment by mjeanroy** [comment]
Awesome, thanks @geoffdutton!
I made some comments, nothing serious, feel free to answer if you disagree.
Please, let me know if you don't have time to update your PR, I'll do it.

**Comment by mjeanroy** [comment]
Hi @geoffdutton,

I would like to thank your for your works, that's awesome! Sadly, I can't merge that for these reasons:
- Your PR contains too much things. I think that it's important to merge features one by one.
- I made a very big refactoring recently to migrate the entire project to ES6 code, there are some conflicts now that needs to be fixed.
- You incremented the version and I would like to update it once a new release is published.

I really appreciate what you did here. If you agree, I can cherry-pick some of your commits ?

File: package.json (hunk 3 above)
Comment (geoffdutton) [inline package.json]: Yep, will do.

File: package.json (hunk 5 above)
Comment (geoffdutton) [inline package.json]: Yep.

File: src/jasmine-utils.js (hunk 6 above)
Comment (geoffdutton) [inline src/jasmine-utils.js]: Yes, I will separate that out.

**Comment by geoffdutton** [comment]
I agree it contains too much. After using it on my own projects, the features I thought would be helpful aren't be used as much, so let me re-evaluate there.

I will fix the conflicts.

And yes, I will revert the version number to what it was. Not sure what I was thinking there ;)

I'll update early next week, I'm hoping to get to this over the weekend. Thanks!

**Comment by geoffdutton** [comment]
I'm going to just open another pull request.

**Pull request closed without merging**